src/components/modes/teacher/Settings.js

```diff
@@ -118,6 +121,11 @@ class Settings extends Component {
     this.setState({ timeControlsVisible: !timeControlsVisible });
   };
 
+  handleChangeTimerVisibility = () => {
+    const { timerVisible } = this.state;
+    this.setState({ timerVisible: !timerVisible });
```

Comment:
In here you should have a clause where if `timerVisible` (i.e. if you're going to change `timerVisible` to `false`, you should make sure that `headerVisible` is `false`.

Reply:
When `timerVisible` is `false`, 
- both header & timer no longer visible.  
- `headerVisible` is turned off and control disabled
- `startImmediately` is turned on and control disabled
- `timeControlsVisible`  removed from the setting.

![Screenshot 2020-05-28 at 12 19 22](https://user-images.githubusercontent.com/18397749/83129797-9ff3e800-a0dd-11ea-83fa-fd991ba7d2b1.png)